app/src/main/proto/vss.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+option java_multiple_files = true;
+package org.vss;
+
+message GetObjectRequest {
+  /*
+    All APIs operate within a single storeId.
+    StoreId is keyspace identifier.
+    Ref: https://en.wikipedia.org/wiki/Keyspace_(distributed_data_store)
+    It is up to clients to use single or multiple store's for their usage.
+   */
+  string storeId = 1;
+  string key = 2;
+}
+message GetObjectResponse {
+  KeyValue value = 2;
+}
+
+message PutObjectRequest {
+  string storeId = 1;
+  /*
+    If present, write will only succeed if current DB globalVersion against a storeId is
+    same as in request.
+    Client is expected to send their current global version at hand.
+
+    For first write of store, version should be '0'. If write succeeds, clients must increment
+    their global version(on device) by 1.
+
+    Requests with conflicting version will fail with [`ConflictException`] as ErrorCode.
```

Comment:
Will this return the actual global version on failure? Or does the client need to do something to get in sync?

Reply:
No, it does not. Returning it from here doesn't add much value and keeps api simpler (and can have some race-conditions if used without considering some edge cases.)
ConflictException indicates that their view of store is out-of-sync. And they need to re-sync their view by fetching all key-versions and global version from server. 
And for the keys having different version on server, client will need to update their view(can be done lazy).
(This is will also be provided as implemented method in vss-client)

Reply:
Is there a method for fetching everything? A client may not have all keys if another client wrote a new key.

Reply:
Yes it will be in subsequent PR once we have get and put.